u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java

```diff
@@ -111,9 +111,11 @@ private JsonArray getTransportsAsJson() {
     return transportsArray;
   }
 
-  public JsonObject getJson() {
+  public JsonObject getJson(String defaultAppId) {
     JsonObject result = new JsonObject();
-    result.addProperty("appId", appId);
+    if (appId != null && !appId.equals(defaultAppId)) {
```

Comment:
Right. Which sounds like a bug, no? In that case, we'll set an invalid appId, which the client will reject. How defensive do we need to be against all possible invalid values? I'd argue there are lots of other ways the server can shoot itself in the foot, and fixing them all is off topic for this pull request.